storage/innobase/handler/handler0alter.cc

```diff
@@ -5761,7 +5762,9 @@ static bool innobase_instant_try(
 		index->clear_instant_alter();
 		goto func_exit;
 	} else if (!user_table->is_instant()) {
-		ut_ad(!user_table->not_redundant());
+		ut_ad(!user_table->not_redundant()
+		      || ha_alter_info->handler_flags
+			     & ALTER_COLUMN_UNVERSIONED);
```

Comment:
I think that there should be parentheses around the `&` operation in order to avoid a warning about operator precedence.
Could we also handle column renaming in this code path?